controlplane/kubeadm/internal/controllers/status.go

```diff
@@ -549,53 +573,116 @@ func setAvailableCondition(_ context.Context, kcp *controlplanev1.KubeadmControl
+			Type:    controlplanev1.KubeadmControlPlaneAvailableV1Beta2Condition,
+			Status:  metav1.ConditionTrue,
+			Reason:  controlplanev1.KubeadmControlPlaneAvailableV1Beta2Reason,
+			Message: strings.Join(messages, "\n"),
 		})
 		return
 	}
 
 	messages := []string{}
 	if !kcp.DeletionTimestamp.IsZero() {
-		messages = append(messages, "Control plane metadata.deletionTimestamp is set")
+		messages = append(messages, "* Control plane metadata.deletionTimestamp is set")
 	}
 
 	if !v1beta2conditions.IsTrue(kcp, controlplanev1.KubeadmControlPlaneCertificatesAvailableV1Beta2Condition) {
-		messages = append(messages, "Control plane certificates are not available")
+		messages = append(messages, "* Control plane certificates are not available")
 	}
 
 	if etcdIsManaged && etcdMembersHealthy < etcdQuorum {
 		switch etcdMembersHealthy {
 		case 0:
-			messages = append(messages, fmt.Sprintf("There are no healthy etcd member, at least %d required for etcd quorum", etcdQuorum))
+			messages = append(messages, fmt.Sprintf("* There are no healthy etcd member, at least %d required for etcd quorum", etcdQuorum))
 		case 1:
-			messages = append(messages, fmt.Sprintf("There is 1 healthy etcd member, at least %d required for etcd quorum", etcdQuorum))
+			messages = append(messages, fmt.Sprintf("* There is 1 of %d healthy etcd member, at least %d required for etcd quorum", len(etcdMembers), etcdQuorum))
 		default:
-			messages = append(messages, fmt.Sprintf("There are %d healthy etcd members, at least %d required for etcd quorum", etcdMembersHealthy, etcdQuorum))
+			messages = append(messages, fmt.Sprintf("* There are %d of %d healthy etcd members, at least %d required for etcd quorum", etcdMembersHealthy, len(etcdMembers), etcdQuorum))
 		}
 	}
 
 	if k8sControlPlaneHealthy < 1 {
-		messages = append(messages, "There are no Machines with healthy control plane components, at least 1 required")
+		messages = append(messages, "* There are no Machines with healthy control plane components, at least 1 required")
 	}
 
 	v1beta2conditions.Set(kcp, metav1.Condition{
 		Type:    controlplanev1.KubeadmControlPlaneAvailableV1Beta2Condition,
 		Status:  metav1.ConditionFalse,
 		Reason:  controlplanev1.KubeadmControlPlaneNotAvailableV1Beta2Reason,
-		Message: strings.Join(messages, ";"),
+		Message: strings.Join(messages, "\n"),
 	})
 }
 
+// shouldSurfaceWhenAvailableTrue defines when a control plane components/etcd issue should surface when
+// Available condition is true.
+// The main goal of this check is to avoid to surface false negatives/flakes, and thus:
+// - issues happening before the node exists are not surfaced (machine is still provisioning)
+// - issues existing for less than 2 minutes are not surfaced (to prevent the message to flick)
+// Note: Two minutes is the time after which we assume that the issue being reported is an actual problem.
+func shouldSurfaceWhenAvailableTrue(machine *clusterv1.Machine, conditionTypes ...string) bool {
```

Comment:
Could be good to have some unit testing for this func specifically

Reply:
done